### PR TITLE
[PINK-252] Add rubocop

### PR DESCRIPTION
**Pull request by kaizencodes**

### CODE REVIEW

File: lib/signable/version.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
 module Signable
-  VERSION = "0.0.7"
+  VERSION = '0.0.8'
```
Comment (christianball) [inline lib/signable/version.rb]: We were discussing semantic versioning the other day.
Should we start using it now do you think?
If the third digit is for bug-fixes, and the second digit is for additions without breaking changes, wouldn't adding Rubocop be a second digit change?
So `0.1.7` ?   Or did I misunderstand?

File: spec/signable/query/response_spec.rb [hunk 2]
```diff
@@ -1,34 +1,40 @@
...
 
-    context "when code is 200" do
-      let(:http_response) { double 'http_response', code: 200 }
-      it { should be true }
+    let(:response) { described_class.new http_response }
+
+    context 'when code is 200' do
+      let(:http_response) { double code: 200 }
+
+      it { is_expected.to be true }
```
Comment (christianball) [inline spec/signable/query/response_spec.rb]:
```suggestion
      it { is_expected.to eq true }
```
For this reason: https://github.com/smartpension/api/pull/12217#discussion_r421541928

File: spec/signable/query/response_spec.rb [hunk 3]
```diff
@@ -1,34 +1,40 @@
...
     end
 
-    context "when code is 202" do
-      let(:http_response) { double 'http_response', code: 202 }
-      it { should be true }
+    context 'when code is 202' do
+      let(:http_response) { double code: 202 }
+
+      it { is_expected.to be true }
```
Comment (christianball) [inline spec/signable/query/response_spec.rb]:
```suggestion
      it { is_expected.to eq true }
```

File: spec/signable/query/response_spec.rb [hunk 4]
```diff
@@ -1,34 +1,40 @@
...
     end
 
-    context "when code is !200" do
-      let(:http_response) { double 'http_response', code: 205 }
-      it { should be false }
+    context 'when code is !200' do
+      let(:http_response) { double code: 205 }
+
+      it { is_expected.to be false }
```
Comment (christianball) [inline spec/signable/query/response_spec.rb]:
```suggestion
      it { is_expected.to eq false }
```

File: spec/signable/query/response_spec.rb [hunk 5]
```diff
@@ -1,34 +1,40 @@
...
+    context 'when code is 202' do
+      let(:http_response) { double code: 202 }
+
+      it { is_expected.to be true }
     end
 
-    context "when code is !200" do
-      let(:http_response) { double 'http_response', code: 205 }
-      it { should be false }
+    context 'when code is !200' do
```
Comment (christianball) [inline spec/signable/query/response_spec.rb]:
```suggestion
    context 'when code is not 200' do
```
Clearer do you think?

File: spec/support/concerns/model.rb [hunk 6]
```diff
@@ -67,32 +66,33 @@ def form_data
...
       described_class.column :required, presence: true
     end
 
-    context "when all required fields are present" do
-      let (:described) { described_class.new required: 'test' }
+    context 'when all required fields are present' do
+      let(:described) { described_class.new required: 'test' }
 
-      it { should be true }
+      it { is_expected.to be true }
```
Comment (christianball) [inline spec/support/concerns/model.rb]:
```suggestion
      it { is_expected.to eq true }
```

File: spec/support/concerns/model.rb [hunk 7]
```diff
@@ -67,32 +66,33 @@ def form_data
...
     end
 
-    context "when all required fields are not present" do
-      let (:described) { described_class.new }
-      it { should be false }
+    context 'when all required fields are not present' do
+      let(:described) { described_class.new }
+
+      it { is_expected.to be false }
```
Comment (christianball) [inline spec/support/concerns/model.rb]:
```suggestion
      it { is_expected.to eq false }
```

File: spec/support/concerns/model.rb [hunk 8]
```diff
@@ -67,32 +66,33 @@ def form_data
...
     end
   end
 
-  describe ".prefix" do
+  describe '.prefix' do
     subject { described_class.prefix }
 
-    it { should be_eql described_class.name.demodulize.underscore }
+    it { is_expected.to be_eql described_class.name.demodulize.underscore }
```
Comment (christianball) [inline spec/support/concerns/model.rb]:
```suggestion
    it { is_expected.to eq described_class.name.demodulize.underscore }
```

File: spec/support/concerns/prefix.rb [hunk 9]
```diff
@@ -1,43 +1,47 @@
...
+  describe '#match?' do
     subject { described.match?(name) }
-    context "when name match name with prefix" do
-      let(:name) { 'prefix_test'}
-      it { should be true }
+
+    context 'when name match name with prefix' do
+      let(:name) { 'prefix_test' }
+
+      it { is_expected.to be true }
```
Comment (christianball) [inline spec/support/concerns/prefix.rb]:
```suggestion
      it { is_expected.to eq true }
```

File: spec/support/concerns/prefix.rb [hunk 10]
```diff
@@ -1,43 +1,47 @@
...
     end
 
-    context "when name match name without prefix" do
-      let(:name) { 'test'}
-      it { should be true }
+    context 'when name match name without prefix' do
+      let(:name) { 'test' }
+
+      it { is_expected.to be true }
```
Comment (christianball) [inline spec/support/concerns/prefix.rb]:
```suggestion
      it { is_expected.to eq true }
```

File: spec/support/concerns/prefix.rb [hunk 11]
```diff
@@ -1,43 +1,47 @@
...
     end
 
-    context "when name do not match" do
-      let(:name) { 'something'}
-      it { should be false }
+    context 'when name do not match' do
+      let(:name) { 'something' }
+
+      it { is_expected.to be false }
```
Comment (christianball) [inline spec/support/concerns/prefix.rb]:
```suggestion
      it { is_expected.to eq false }
```

File: spec/support/concerns/prefix.rb [hunk 12]
```diff
@@ -1,43 +1,47 @@
...
+    context 'when name match name without prefix' do
+      let(:name) { 'test' }
+
+      it { is_expected.to be true }
     end
 
-    context "when name do not match" do
-      let(:name) { 'something'}
-      it { should be false }
+    context 'when name do not match' do
```
Comment (christianball) [inline spec/support/concerns/prefix.rb]:
```suggestion
    context 'when name does not match' do
```

File: spec/support/concerns/prefix.rb [hunk 13]
```diff
@@ -1,43 +1,47 @@
...
+      it { is_expected.to be false }
     end
   end
 
-  describe "#name_with_prefix" do
+  describe '#name_with_prefix' do
     subject { described.name_with_prefix }
 
-    it { should be_eql 'prefix_test' }
+    it { is_expected.to be_eql 'prefix_test' }
```
Comment (christianball) [inline spec/support/concerns/prefix.rb]:
```suggestion
    it { is_expected.to eq 'prefix_test' }
```

File: spec/support/concerns/prefix.rb [hunk 14]
```diff
@@ -1,43 +1,47 @@
...
   end
 
-  describe "#prefix" do
+  describe '#prefix' do
     subject { described.prefix }
 
-    it { should be_eql 'prefix' }
+    it { is_expected.to be_eql 'prefix' }
```
Comment (christianball) [inline spec/support/concerns/prefix.rb]:
```suggestion
    it { is_expected.to eq 'prefix' }
```

File: spec/support/concerns/prefix.rb [hunk 15]
```diff
@@ -1,43 +1,47 @@
...
-    let(:described) { described_class.new 'prefix_test', { prefix: 'prefix' } }
-
+  describe '#name_without_prefix' do
     subject { described.name_without_prefix }
 
-    it { should be_eql 'test' }
-  end
+    let(:described) { described_class.new 'prefix_test', { prefix: 'prefix' } }
 
+    it { is_expected.to be_eql 'test' }
```
Comment (christianball) [inline spec/support/concerns/prefix.rb]:
```suggestion
    it { is_expected.to eq 'test' }
```

File: spec/support/concerns/query.rb [hunk 16]
```diff
@@ -1,70 +1,74 @@
...
 
-  describe "#save" do
-    context "when persisted" do
+  describe '#save' do
+    context 'when persisted' do
       let(:described) { described_class.new id: 1 }
 
-      it "call client update" do
-        expect(described_class.client).to receive(:update).with(described_class.entry_point, 1, described).and_return response
+      it 'call client update' do
```
Comment (christianball) [inline spec/support/concerns/query.rb]:
```suggestion
      it 'calls update on the client' do
```

File: spec/support/concerns/query.rb [hunk 17]
```diff
@@ -1,70 +1,74 @@
...
       end
     end
 
-    context "when not persisted" do
+    context 'when not persisted' do
       let(:described) { described_class.new }
 
-      it "call client create" do
-        expect(described_class.client).to receive(:create).with(described_class.entry_point, described).and_return response
+      it 'call client create' do
```
Comment (christianball) [inline spec/support/concerns/query.rb]:
```suggestion
      it 'calls create on the client' do
```

File: spec/support/concerns/query.rb [hunk 18]
```diff
@@ -1,70 +1,74 @@
...
     end
   end
 
-  describe "#delete" do
+  describe '#delete' do
     let(:described) { described_class.new id: 1 }
 
-    it "call client delete" do
-      expect(described_class.client).to receive(:delete).with(described_class.entry_point, described.id).and_return response
+    it 'call client delete' do
```
Comment (christianball) [inline spec/support/concerns/query.rb]:
```suggestion
    it 'calls delete on the client' do
```

File: spec/support/concerns/query.rb [hunk 19]
```diff
@@ -1,70 +1,74 @@
...
+  describe '#persisted?' do
+    subject { described.persisted? }
 
-    subject { described.persisted?}
+    let(:described) { described_class.new }
 
-    context "when id is nil" do
-      it { should be false }
+    context 'when id is nil' do
+      it { is_expected.to be false }
```
Comment (christianball) [inline spec/support/concerns/query.rb]:
```suggestion
      it { is_expected.to eq false }
```

File: spec/support/concerns/query.rb [hunk 20]
```diff
@@ -1,70 +1,74 @@
...
     end
 
-    context "when id is present" do
+    context 'when id is present' do
       before do
         described.id = 1
       end
-      it { should be true }
+
+      it { is_expected.to be true }
```
Comment (christianball) [inline spec/support/concerns/query.rb]:
```suggestion
      it { is_expected.to eq true }
```

File: spec/support/concerns/query.rb [hunk 21]
```diff
@@ -1,70 +1,74 @@
...
     end
   end
 
-  describe ".all" do
-    it "call client all with default value for offset and limit" do
+  describe '.all' do
+    it 'call client all with default value for offset and limit' do
```
Comment (christianball) [inline spec/support/concerns/query.rb]:
```suggestion
    it 'calls all on the client with default values for offset and limit' do
```

File: spec/support/concerns/query.rb [hunk 22]
```diff
@@ -1,70 +1,74 @@
...
       expect(described_class.client).to receive(:all).with(described_class.entry_point, 0, 30).and_return response
       described_class.all
     end
   end
 
-  describe ".find" do
-    it "call client find" do
+  describe '.find' do
+    it 'call client find' do
```
Comment (christianball) [inline spec/support/concerns/query.rb]:
```suggestion
    it 'calls find on the client' do
```

File: lib/signable/version.rb (hunk 1 above)
Comment (kaizencodes) [reply lib/signable/version.rb]: no in this case it would be 0.1.0.

Comment (christianball) [reply lib/signable/version.rb]: Shall we do that then?

Comment (kaizencodes) [reply lib/signable/version.rb]: let's 489079f

File: spec/signable/query/response_spec.rb (hunk 2 above)
Comment (kaizencodes) [reply spec/signable/query/response_spec.rb]: f139680

File: spec/signable/query/response_spec.rb (hunk 3 above)
Comment (kaizencodes) [reply spec/signable/query/response_spec.rb]: f139680

File: spec/signable/query/response_spec.rb (hunk 4 above)
Comment (kaizencodes) [reply spec/signable/query/response_spec.rb]: f139680

**Review by christianball — APPROVED** [review]

**Review by bugthing — APPROVED** [review]

**Pull request merged**